### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by overlordbleck**
**[Tested/Finished]**  ~~[Untested/Working branch]~~
Since user messages were deprecated I decided to rework the entire addon (giving it more descriptive disconnect messages too!).

### CODE REVIEW

**Comment by plally** [comment]
might want to use more unique filenames, ive had issues using names like sv_init when another addon also uses that same filename with the same path

**Comment by sarahsturgeon** [comment]
Was writing literally the same thing ^ Better safe than sorry

**Comment by plally** [comment]
shouldn't ` lua/autorun/cl/` be` lua/autorun/client/`, same with `autorun/sv/`

**Comment by overlordbleck** [comment]
~~I don't think it matters, pretty sure both work. (I would confirm this but server isnt working)~~
Yes you're right...

**Comment by overlordbleck** [comment]
100% functional now ~~, I need to find a solution to get the players team when they disconnect with this method.~~

**Comment by sarahsturgeon** [comment]
I'll leave a review tonight or tomorrow - I have some comments about code style

File: lua/autorun/server/sv_apple.lua [hunk 1]
```diff
@@ -29,12 +29,18 @@ end
...
     local name = data.name
     local steamID = data.networkid
     local userID = data.userid
     local isBot = data.bot
     local reason = data.reason
-    local plyTeam = Player( userID ):Team()
+
+    if IsValid( ply ) then
+        plyTeam = Player( userID ):Team()
+    end
```
Comment (sarahsturgeon) [inline lua/autorun/server/sv_apple.lua]:
```lua
local plyTeam = IsValid( ply ) and Player( userID ):Team() or TEAM_UNASSIGNED
```

🤓 🧐

**Review by sarahsturgeon — APPROVED** [review]

File: lua/autorun/server/sv_apple.lua (hunk 1 above)
Comment (overlordbleck) [reply lua/autorun/server/sv_apple.lua]: totally forgot you could do 1 liners with lua, good eye

File: lua/autorun/server/sv_apple.lua [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        plyTeam = ply:Team()
+    end
+
+    MsgN( "Player " .. name .. " has left the server. (" .. reason .. ")" )
+
+    net.Start( "cfc_playerdisconnect" )
+        net.WriteString( name )
+        net.WriteString( steamID )
+        net.WriteString( reason )
+        net.WriteInt( plyTeam, 11 )
```
Comment (plally) [inline lua/autorun/server/sv_apple.lua:45]: any reason you chose 11 bits?

**Review by plally — APPROVED** [review]

File: lua/autorun/server/sv_apple.lua (hunk 2 above)
Comment (overlordbleck) [reply lua/autorun/server/sv_apple.lua:45]: Team ID's go from 0 to 1002, where 1001 is `TEAM_UNASSIGNED` and 1002 is `TEAM_SPECTATOR`. 11 bits we get a maximum value of 1023.

**Comment by overlordbleck** [comment]
Should be ready for a merge whenever you think its ready @brandonsturgeon

**Pull request merged**